### PR TITLE
Added comments to track history of investigation about shader resourc…

### PR DESCRIPTION
**Pull request by santorac**
…e visibility on vulkan.

It's only a comment and a factored out variable, but I ran AtomSampleViewer full test suite on vulkan.

Signed-off-by: santorac <55155825+santorac@users.noreply.github.com>

### CODE REVIEW

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/DescriptorSetLayout.cpp [hunk 1]
```diff
@@ -159,6 +159,30 @@ namespace AZ
...
+            //   SceneSrg (and ViewSrg(s)) and uses this instance for many shaders with different resource visibilities.
+            // - Same for RayTracingSrgs.shader's "RayTracingSceneSrg" and "RayTracingMaterialSrg"
+            // - Same for ForwardPassSrg.shader's "PassSrg". (This one is especially problematic because, unlike the above cases, we can't just add some special
+            //   handling for the particular SRG name; "PassSrg" is widely used as the name for many different per-pass SRG layouts).
+            // - ShaderResourceGroupPool is intentionally set up to reuse SRGs regardless of visibility, per ShaderResourceGroup::MakeInstanceId which uses
+            //   the source file path in the unique ID (ex: "D:\o3de\Gems\Atom\RPI\Assets\ShaderLib\Atom\RPI\ShaderResourceGroups\DefaultDrawSrg.azsli")
+            //   so that any shader that uses this azsli file will share the same pool and thus share the same PipelineLayoutDescriptor.
+            // In order to address the above issues, we would need to update AZSLc to support some kind of attribute by which the shader-author can manually
```
Comment (moudgils) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/DescriptorSetLayout.cpp:177]: Suggestion
"In order to address the above issues, we would need to update AZSLc.." -> One possible solution could be to update AZSLc..

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/DescriptorSetLayout.cpp [hunk 2]
```diff
@@ -159,6 +159,30 @@ namespace AZ
...
+            // override the visibility for each resource, or we add some new metadata to the .shader files to provide explicit overrides for particular resource
+            // visibilities. Either way, this would likely become error prone and difficult to maintain.
+            // 
+            // There are even more issues beyond what's described above; I prototyped some code to address them and I was still getting RHI validation
+            // failures. I had to give up and move on, so I don't know how deep this rabbit hole goes.
```
Comment (moudgils) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/DescriptorSetLayout.cpp:182]: I would remove this comment.  Instead consider saying

"It is possible that even with the proposed solution above we could run into other issues not foreseen here. Instead a better path forward would be to see if Khronos can address the api requirement itself. A ticket has been opened with Khronos here - https://github.com/KhronosGroup/Vulkan-Docs/issues/1790.
For context the driver error when the visibility bit from PSO does not match the one from Descriptor set is as follows ->

**Review by moudgils — APPROVED** [review]
Approved but I have recommendations for the comments

**Review by Adi-Amazon — APPROVED** [review]

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/DescriptorSetLayout.cpp [hunk 3]
```diff
@@ -159,6 +159,31 @@ namespace AZ
             const AZStd::span<const RHI::ShaderInputSamplerDescriptor> samplerDescs = m_shaderResourceGroupLayout->GetShaderInputListForSamplers();
             const AZStd::span<const RHI::ShaderInputStaticSamplerDescriptor>& staticSamplerDescs = m_shaderResourceGroupLayout->GetStaticSamplers();
 
+            // About VK_SHADER_STAGE_ALL...
+            // We attempted to configure the descriptor set with the actual resource visibility but it was problematic...
```
Comment (Adi-Amazon) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/DescriptorSetLayout.cpp:163]: Suggest TLDR due to length - perhaps single sentence of the problem and current solution, and then dive into the long description.

**Pull request merged**